### PR TITLE
Remove old drivers

### LINKED ISSUE #616: Move Dycoms radiation specs to driver
Including `DYCOMSRadiation`

### PR DESCRIPTION
**Pull request by kpamnany**
~~Based on #592 but does not include the Rayleigh Benard part~~. Updated version of #592.

This doesn't work. @akshaysridhar: need your help on this one!

Closes #616.

### CODE REVIEW

**Comment by akshaysridhar** [comment]
@kpamnany
https://github.com/climate-machine/CLIMA/tree/as/cleanup-kp has the changes / fixes you requested. 
1) The `problem-setup` is now moved out from the `test/DGmethods/compressible_Navier_Stokes/`... directory into `examples/Atmos/ ` and makes use of the Driver-Configuration framework

**Comment by kpamnany** [comment]
Does `dry_rayleigh_benard.jl` not qualify as an 'experiment' that goes in that directory under LES/GCM?

This might be a good thing to determine now, i.e. the meaning of the locations. My interpretation:

1. `test/` contains code that tests various CLIMA features, for CI.
2. `examples/` contains tutorial type code, for introducing the use of CLIMA.
3. `experiments/` contains actual science experiments that produce (potentially publishable) results.

The latter two could possibly be merged? I was thinking that `examples/` would eventually be removed once everything there is migrated over to the CLIMA driver but perhaps the distinction is useful?

Cc: @simonbyrne

**Comment by akshaysridhar** [comment]
1) It's an idealised test problem showing use of the driver-config setup, in line with `2.` above, not truly an `Atmos-LES` `/experiments/...` (which I imagine would include `dycoms`, upcoming `cfsites` , `bomex / trmm / gables` etc. (which are atmospheric LES in pycles)

**Comment by charleskawczynski** [comment]
I like the break-down into 3 sections as @kpamnany suggested! Maybe we can further specify that `test/` is for verification (are we solving the equations correctly?) while `experiments` is for validation (are we solving the correct equations?).

**Comment by kpamnany** [comment]
In that case, is `experiments/GCM/heldsuarez.jl` an 'experiment' or an 'example'?

**Comment by kpamnany** [comment]
@akshaysridhar: I have merged your branch (`as/cleanup-kp`) into this (so you can delete that branch) and squashed all commits. However, #660 was merged yesterday and has changed how boundary conditions are implemented in master. Will you take a look? `git rebase origin/master` in this branch produces conflicts in ` src/Atmos/Model/boundaryconditions.jl` which need to be resolved, and the BCs that were just moved out need to be rechecked.

**Comment by akshaysridhar** [comment]
> @akshaysridhar: I have merged your branch (`as/cleanup-kp`) into this (so you can delete that branch) and squashed all commits. However, #660 was merged yesterday and has changed how boundary conditions are implemented in master. Will you take a look? `git rebase origin/master` in this branch produces conflicts in ` src/Atmos/Model/boundaryconditions.jl` which need to be resolved, and the BCs that were just moved out need to be rechecked.

Noted. Will sort this out today.

**Comment by kpamnany** [comment]
See also #684.

**Comment by kpamnany** [comment]
@akshaysridhar: I think all that remains here is to review [this](https://github.com/climate-machine/CLIMA/pull/672/files#diff-ced05f4047c4b34c23dc44e77babc930R127-R282) commented code to extract any changes that may have come from #684 from here into the respective driver files, and then delete the commented code.

**Comment by akshaysridhar** [comment]
> @akshaysridhar: I think all that remains here is to review [this](https://github.com/climate-machine/CLIMA/pull/672/files#diff-ced05f4047c4b34c23dc44e77babc930R127-R282) commented code to extract any changes that may have come from #684 from here into the respective driver files, and then delete the commented code.

New b.c.s from #  and #684 are merged locally - addressing those comments prior to push.

File: slurmci-test.toml [hunk 1]
```diff
@@ -9,20 +9,12 @@ cpu = [
...
-  { file = "examples/Microphysics/ex_1_saturation_adjustment.jl", slurmargs = ["--ntasks=3"], args = [] },
-  { file = "examples/Microphysics/ex_2_Kessler.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/Euler/isentropicvortex.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/Euler/isentropicvortex-imex.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/Euler/isentropicvortex-multirate.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/Euler/acousticwave-1d-imex.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/compressible_Navier_Stokes/mms_bc_atmos.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "test/DGmethods/compressible_Navier_Stokes/mms_bc_dgmodel.jl", slurmargs = ["--ntasks=3"], args = [] },
-  { file = "test/DGmethods/compressible_Navier_Stokes/rising_bubble-model.jl", slurmargs = ["--ntasks=3"], args = [] },
-  { file = "test/DGmethods/compressible_Navier_Stokes/rising_bubble-model-imex.jl", slurmargs = ["--ntasks=3"], args = [] },
```
Comment (simonbyrne) [inline slurmci-test.toml]: Add new RTB experiments?

**Comment by kpamnany** [comment]
bors try

**Comment by simonbyrne** [comment]
very nice!

**Comment by akshaysridhar** [comment]
bors try

**Comment by kpamnany** [comment]
Ack. 28 commits?

**Comment by kpamnany** [comment]
bors try-

**Comment by kpamnany** [comment]
Replacing with #694.

**Pull request closed without merging**